src/scripts.c

```diff
@@ -34,45 +34,59 @@
 
 static void update_kernel(void)
 {
-	char *kernel_update_cmd = NULL;
-
 	if (strcmp("/", path_prefix) == 0) {
-		string_or_die(&kernel_update_cmd, "/usr/bin/kernel_updater.sh");
+		char *const kernel_update_cmd[] = { "/usr/bin/kernel_updater.sh", NULL } ;
+
+		system_argv(kernel_update_cmd);
```

Comment:
Since the return value is discarded, please use the (void) cast for this function. Similarly for the other call sites below.